subiquitycore/tui.py

```diff
@@ -286,6 +290,37 @@ def extra_urwid_loop_args(self):
     def make_screen(self, inputf=None, outputf=None):
         return make_screen(self.opts.ascii, inputf, outputf)
 
+    def get_initial_rich_mode(self) -> bool:
+        """Return the initial value for rich-mode, either loaded from an
+        exising state file or automatically determined. True means rich mode
+        and False means basic mode."""
+        if self.opts.run_on_serial:
+            rich_mode_file = "rich-mode-serial"
+        else:
+            rich_mode_file = "rich-mode-tty"
+        try:
+            fp = open(self.state_path(rich_mode_file))
+        except FileNotFoundError:
+            pass
+        else:
+            with fp:
+                return json.load(fp)
+
+        try:
+            # During the 23.10 development cycle, there was only one rich-mode
+            # state file. Let's handle the scenario where we just snap
+            # refresh-ed from a pre 23.10 release.
+            # Once mantic is EOL, let's remove this code.
+            fp = open(self.state_path("rich-mode"))
```

Comment:
honestly my opinion is that we don't need to bother with this. but you've done the work now so fine!